src/meta-srv/src/election/mysql.rs

```diff
@@ -0,0 +1,1373 @@
+        if self.is_leader.load(Ordering::Relaxed) {
+            Ok(self.leader_value.as_bytes().into())
+        } else {
+            let key = self.election_key();
+            if let Some((leader, expire_time, current, _)) =
+                self.get_value_with_lease(&key, false).await?
+            {
+                ensure!(expire_time > current, NoLeaderSnafu);
+                Ok(leader.as_bytes().into())
+            } else {
+                NoLeaderSnafu.fail()
+            }
+        }
+    }
+
+    async fn resign(&self) -> Result<()> {
+        todo!()
+    }
+
+    fn subscribe_leader_change(&self) -> broadcast::Receiver<LeaderChangeMessage> {
+        self.leader_watcher.subscribe()
+    }
+}
+
+impl MySqlElection {
+    async fn query(&self, query: Query<'_, MySql, MySqlArguments>) -> Result<Vec<MySqlRow>> {
+        let mut client = self.client.lock().await;
+        let res = query
+            .fetch_all(&mut *client)
+            .await
+            .context(MySqlExecutionSnafu)?;
+        Ok(res)
+    }
+
+    async fn execute(&self, query: Query<'_, MySql, MySqlArguments>) -> Result<u64> {
+        let mut client = self.client.lock().await;
+        let res = query
+            .execute(&mut *client)
+            .await
+            .context(MySqlExecutionSnafu)?;
+        Ok(res.rows_affected())
+    }
+
+    /// Returns value, expire time and current time. If `with_origin` is true, the origin string is also returned.
+    async fn get_value_with_lease(
+        &self,
+        key: &str,
+        with_origin: bool,
+    ) -> Result<Option<(String, Timestamp, Timestamp, Option<String>)>> {
+        let key = key.as_bytes();
+        let query = sqlx::query(&self.sql_set.get_value_with_lease).bind(key);
+        let res = self.query(query).await?;
+
+        if res.is_empty() {
+            Ok(None)
+        } else {
+            // Safety: Checked if res is empty above.
+            let current_time_str = String::from_utf8_lossy(res[0].try_get(1).unwrap());
+            let current_time = match Timestamp::from_str(&current_time_str, None) {
+                Ok(ts) => ts,
+                Err(_) => UnexpectedSnafu {
+                    violated: format!("Invalid timestamp: {}", current_time_str),
+                }
+                .fail()?,
+            };
+            // Safety: Checked if res is empty above.
+            let value_and_expire_time =
+                String::from_utf8_lossy(res[0].try_get(0).unwrap_or_default());
+            let (value, expire_time) = parse_value_and_expire_time(&value_and_expire_time)?;
+
+            if with_origin {
+                Ok(Some((
+                    value,
+                    expire_time,
+                    current_time,
+                    Some(value_and_expire_time.to_string()),
+                )))
+            } else {
+                Ok(Some((value, expire_time, current_time, None)))
+            }
+        }
```

Comment:
Nit(but non-blocking): I'd like to use B instead of A because it reduces one tab indentation for most of the code.
A: 
```
if res.is_empty() {
    Ok(None)
} else {
    // other ....
}
```

B:
```
if res.is_empty() {
    return Ok(None);
}

// other ....
```